### ISSUE TITLE
Workspace indicator disappeared

### ISSUE THREAD

**Issue by joshwisehub** [SolDoesTech/HyprV4#59]
I recently updated Hyprland and the issue occurred. Am currently on the latest version
{
    "branch": "",
    "commit": "96d555e8e794627bfc561e294e148ab8a9961fcc",
    "dirty": true,
    "commit_message": "",
    "tag": "v0.29.1",
    "flags": []
}

**Comment by jontstaz** [comment]
When you say "Workspace Indicator" do you mean the "1", "2", "3", etc.. that was previously showing at the top of the screen to indicate what desktop/workspace you're on/how many are active? Because I'm facing the same exact issue. For the past week all of a sudden it disappeared and I can't seem to get it back no matter what I try. It's really annoying. The icons still display at the bottom but the workspace numbers have gone away. Not just for v4 but for all of them.

**Comment by stove-jeebs** [comment]
![image](https://github.com/SolDoesTech/HyprV4/assets/67490238/a762cbc0-481c-476f-ac3c-ea93b0db17f4)

new hyprland update no longer supports `wlr/workspaces`

**Comment by joshwisehub** [comment]
yes i fixed it .By removing wlr/workspace and it now works perfectly fine

**Comment by AtharvaT02** [comment]
Hello Josh, I just installed hyperv4 and am facing the same issue. can you tell me step by step what I should do. Im very new to linux

**Comment by joshwisehub** [comment]
the issue sis solved by changing waybar config files. here is the link
https://github.com/SolDoesTech/HyprV4/pull/55/files.Available on the pull
requests

**Comment by chdrsto** [comment]
@joshwisehub , i was able to get the workspaces on the screen by simple replacing "wlr" with "hyprland" ... means, from `wlr/workspaces` in `hyprland/workspaces`

in /home/.config/HyprV/waybar/conf/v4-config.jsonc (as well as in v1, v2, v3)

the easiest way is to use sed ...like :
`sed -i 's/wlr/hyprland/g' v4-config.jsonc`

My only issue at the moment is, that I see all 10 workspaces. This I need to figure out how to show only active ones

**Comment by t4fita** [comment]
@chdrsto  Thanks for your answer it was the easiest one I found that solved my problem ! 
 
> My only issue at the moment is, that I see all 10 workspaces. This I need to figure out how to show only active ones

As for this, the solution I found was just to reduce the number of workspaces. This is not really solving the problem as what you want is only to show the active ones, but it's a lot more pleasing to see only a few workspaces ! 

I believe you can achieve that by modifying the same file ` /home/.config/HyprV/waybar/conf/v4-config.jsonc` (as well as in v1, v2, v3) and in persistent_workspaces, you can reduce the number of workspaces by just deleting these. So for 5 workspaces you should seeyou should see something like:
```
 "hyprland/workspaces": {
        "disable-scroll": true,
        "all-outputs": true,
        "on-click": "activate",
        "persistent_workspaces": {
            "1": [],
            "2": [],
            "3": [],
            "4": [],
            "5": []
        }
    },

```

**Comment by hangocdinhnew** [comment]
@t4fita Sorry for being super late, but persistent_workspaces is actually persistent now, so reducing the number of workspaces is not needed, you just remove the persistent_, it's that simple.

EDIT: Something like this:

```
"hyprland/workspaces": {
        "disable-scroll": true,
        "all-outputs": true,
        "on-click": "activate",
        "workspaces": {
            "1": [],
            "2": [],
            "3": [],
            "4": [],
            "5": [],
            "6": [],
            "7": [],
            "8": [],
            "9": [],
            "10": [] 
        }
 },
```

**Comment by liam-rar** [comment]
> sed -i 's/wlr/hyprland/g' v4-config.jsonc

This worked for me, thank you
